### PR TITLE
Convert progress bar metrics to float

### LINKED ISSUE #5481: Progress Bar on release/1.2-dev contains tensors
## Please reproduce using the BoringModel

```
class LoggerConnector:
    def __init__(self, trainer):
        self.trainer = trainer
        self._callback_metrics = MetricsHolder()
        self._evaluation_callback_metrics = MetricsHolder(to_float=True)
        self._logged_metrics = MetricsHolder()
        self._progress_bar_metrics = MetricsHolder()
```

self._logged_metrics should be `to_float=True` for progress_bar and to_float=False for merging within callbacks.

The internal API needs to be changed.

### To Reproduce

Use following [**BoringModel**](https://colab.research.google.com/drive/1HvWVVTK8j2Nj52qU4Q4YCyzOm0_aLQF3?usp=sharing) and post here

### Environment

**Note**: `Bugs with code` are solved faster ! `Colab Notebook` should be made `public` !

* `IDE`: Please, use our python [bug_report_model.py](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pl_examples/bug_report_model.py
) template.

* `Colab Notebook`: Please copy and paste the output from our [environment collection script](https://raw.githubusercontent.com/PyTorchLightning/pytorch-lightning/master/tests/collect_env_details.py) (or fill out the checklist below manually).

You can get the script and run it with:
```
wget https://raw.githubusercontent.com/PyTorchLightning/pytorch-lightning/master/tests/collect_env_details.py
# For security purposes, please check the contents of collect_env_details.py before running it.
python collect_env_details.py
```

 - PyTorch Version (e.g., 1.0):
 - OS (e.g., Linux):
 - How you installed PyTorch (`conda`, `pip`, source):
 - Build command you used (if compiling from source):
 - Python version:
 - CUDA/cuDNN version:
 - GPU models and configuration:
 - Any other relevant information:

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Fixes #5481

~~### TODO:~~
~~- Separate train/val/test metrics~~
~~- `callback_metrics` should copy the data as is, not as a reference from `progress_bar_metrics`. This way, each `MetricHolder` is free to make their own conversions without affecting other parts.~~

edit: leaving refactor out of this PR

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [n/a] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified
 - [x] **Check that target branch and milestone match!**

### CODE REVIEW

**Review by tchaton — APPROVED** [review]

**Comment by Borda** [comment]
@carmocca is it still WIP or ready to go? :]

**Comment by carmocca** [comment]
So I planned to include a larger refactor here. I think i'll commit the one-line fix to master and work on the refactor here.

**Comment by Borda** [comment]
> So I planned to include a larger refactor here. I think i'll commit the one-line fix to master and work on the refactor here.

I would rather have multiple smaller refactors, it is easier to review :]

File: tests/callbacks/test_progress_bar.py [hunk 1]
```diff
@@ -340,3 +341,25 @@ def test_test_progress_bar_update_amount(tmpdir, test_batches, refresh_rate, tes
     )
     trainer.test(model)
     progress_bar.test_progress_bar.update.assert_has_calls([call(delta) for delta in test_deltas])
+
+
+def test_test_progress_bar_thing(tmpdir):
```
Comment (carmocca) [inline tests/callbacks/test_progress_bar.py:346]:
```suggestion
def test_tensor_to_float_conversion(tmpdir):
```

**Review by s-rog — APPROVED** [review]

**Review by Borda — APPROVED** [review]

**Review by awaelchli — COMMENTED** [review]
Just one request: could we move the test to the logging tests? Because the conversion doesn't happen in the progress bar therefore I propose to move this test outside the progress bar unit tests.

**Review by Borda — CHANGES_REQUESTED** [review]
pls see https://github.com/PyTorchLightning/pytorch-lightning/pull/5692#pullrequestreview-583728293

**Comment by carmocca** [comment]
> Just one request: could we move the test to the logging tests? Because the conversion doesn't happen in the progress bar therefore I propose to move this test outside the progress bar unit tests.

I disagree. The purpose of the test is to check that the progress bar correctly displays the number as a float, not a tensor. The fact that the fix is in the logger connector is independent to what we want to fix. If the logger connector was completely replaced by a different abstraction, I still want to make sure the progress bar does not change.

**Review by awaelchli — APPROVED** [review]
I am not convinced with your argument but it's ok I don't want to block. If we change our mind we can move the test later.

File: tests/callbacks/test_progress_bar.py [hunk 2]
```diff
@@ -349,3 +350,25 @@ def test_test_progress_bar_update_amount(tmpdir, test_batches, refresh_rate, tes
     )
     trainer.test(model)
     progress_bar.test_progress_bar.update.assert_has_calls([call(delta) for delta in test_deltas])
+
+
+def test_tensor_to_float_conversion(tmpdir):
+    """Check tensor gets converted to float"""
+    class TestModel(BoringModel):
```
Comment (carmocca) [inline tests/callbacks/test_progress_bar.py:357]:
```suggestion

    class TestModel(BoringModel):

```

**Pull request merged**